doc/source/train/architecture.rst

```diff
@@ -3,42 +3,61 @@
 Ray Train Architecture
 ======================
 
-A diagram of the Ray Train architecture is provided below.
+The process of training models with Ray Train consists of several components.
+First, depending on the training framework you want to work with, you will have
+to provide a so-called ``Trainer`` that manages the training process.
+For instance, to use a PyTorch model, you use a ``TorchTrainer``.
+The actual training load is distributed among workers on a cluster that belong
+to a ``WorkerGroup``.
+Each framework has its specific communication protocols and exchange formats,
+which is why Ray Train provides ``Backend`` implementations (e.g. ``TorchBackend``)
+that can be used to run the training process using a `BackendExecutor`.
+
+Here's a visual overview of the architecture components of Ray Train:
```

Comment:
Should we rename this guide to be deep learning specific? The boosted tree trainers actually don't have any of these backend/executor/worker group abstractions.

Also, this diagram is severely outdated 😅

Reply:
I know... man, this particular doc is painful to read. That's why I touched it up, but it's a bit of a rabbit hole!

Reply:
I'll add a TODO, but we need to commit to fixing this. I think having a general Architecture Guide is useful to have.